### DISCUSSION TITLE
How do I use this on windows?

### DISCUSSION THREAD

**Discussion by AwesomeSavage** [Darthsae/TCreator#7] [Q&A]
I downloaded the latest release a long time ago, and I have python installed, but all I get is a blank console window that immediately closes when I open the application.

**Comment by AwesomeSavage** [comment]
I know it works (for others), so this isn't a bug, but it is a me problem.

**Comment by AwesomeSavage** [comment]
Are there other dependencies required, besides python? If so what are they, and can you list them in the readme?

**Reply by Darthsae** [reply]
Yes, you will need to have Pillow and Tkinter:
pip install pillow
pip install tk

**Reply by AwesomeSavage** [reply]
Well, I installed those, and the console window opens with some text I can't read, due to the window closing less than a second after it opens.

**Comment by AwesomeSavage** [comment]
What should I do now?

**Reply by Darthsae** [reply]
Did you change the path in settings.txt?

**Reply by Darthsae** [reply]
By default it should say:
C:\Users\sockm\OneDrive\Documents\My Games\Terraria\tModLoader\ModSources {EXAMPLE PATH}

**Reply by Darthsae** [reply]
You need to change that to your mod sources directory, ie like this:
C:\Users\Whatever\Goes\Here

**Reply by AwesomeSavage** [reply]
Yes, I did change the path, that's the first thing I did when I installed the application.

**Reply by Darthsae** [reply]
Hmm, interesting. Do you think you could get me what the terminal says through some method? That might help me find out what problem is occurring, since that seems to be a bug with the program.

**Reply by AwesomeSavage** [reply]
Ok
Is a specific python version required by chance?

**Reply by AwesomeSavage** [reply]
I only have 3.12 right now.

**Reply by Darthsae** [reply]
Maybe, try updating to 3.14 possibly.

**Reply by AwesomeSavage** [reply]
3.14 Doesn't exist. Do you mean 3.13?

**Reply by Darthsae** [reply]
3.11 should have it working, do you have any mods in the directory you have in settings.txt?

**Reply by AwesomeSavage** [reply]
Yes, I have a mod there in the middle of a partly finished update.

**Reply by ArctanDeveloper** [reply]
Ok, that's really weird, is your settings.txt directory only the directory, because in the default install there is a {Example Path} part to show that the path should be changed, also make sure it's on the first line like this:

C:/Users/ExampleUser/Mods

And that's the whole thing.

**Reply by ArctanDeveloper** [reply]
I got it to work by doing that. So that might be something you could try.

**Reply by Darthsae** [reply]
You might have to get me the logs for me to see the error.

**Comment by AwesomeSavage** [comment]
I have no idea how to get console logs, are there logs stored for this program anywhere by chance?
Also, my path is the only thing in the settings.txt.

**Reply by Darthsae** [reply]
No, the logs are in the actual command prompt, if you open the command prompt in the directory and run:
py main.py
You should be able to see the logs, and with those I can find the issue.

**Comment by AwesomeSavage** [comment]
Alright, cool. I actually found the approximate position of the issue by inserting a pause after each procedure and class, for which I edited the file, but I am going to remove it now. The issue I found seems to be with the creation of the main menu, because when the function is called, the window crashes. I put a pause before the call, and after I unpaused, it crashed, and when I put the pause after, it never got there. I will get you the specific information from the logs now.

**Reply by Darthsae** [reply]
Thank you, that is helpful to know, I'll look into it once I have the logs and see if I can do a patch for the current release.

**Reply by AwesomeSavage** [reply]
I got it working, somehow the version of python I had was weird, probably because it was from the Microsoft Store, so I went and installed it from google instead, and had to do a little tweaking, but now it works! Yay!

**Reply by AwesomeSavage** [reply]
I guess that stuff from Microsoft is usually scuffed if it can be obtained elsewhere, so I should remember that.

**Reply by AwesomeSavage** [reply]
Oddly enough, even stuff owned by Microsoft is scuffed half of the time, too. Luckly Minecraft isn't, at least not yet.

**Reply by Darthsae** [reply]
Ok, I'll add in a disclaimer about that then, in the rewrite I'll try and make it an executable so stuff like this can be avoided in the future.

**Reply by AwesomeSavage** [reply]
Alright, thanks! Bye!
